lib/assets/javascripts/turbograft/turbolinks.coffee

```diff
@@ -49,6 +49,7 @@ if browserSupportsCustomEvents
   installJqueryAjaxSuccessPageUpdateTrigger()
 
 replaceNode = (newNode, oldNode) ->
+  triggerEvent('page:before-node-removed', {newNode: newNode, oldNode: oldNode})
```

Comment:
We should probably fire this event inside removeNode as well, considering the name. Otherwise the event should be renamed to `page:before-node-replaced`